core/dbt/parser/manifest.py

```diff
@@ -313,96 +403,50 @@ def reparse_macros(self):
-            return self.manifest.sanitized_update(
-                block.file, self.old_manifest, parser.resource_type
-            )
-        return False
-
     def write_manifest_for_partial_parse(self):
         path = os.path.join(self.root_project.target_path,
                             PARTIAL_PARSE_FILE_NAME)
-        make_directory(self.root_project.target_path)
-        with open(path, 'wb') as fp:
-            pickle.dump(self.manifest, fp)
+        try:
+            manifest_msgpack = self.manifest.to_msgpack()
+            make_directory(os.path.dirname(path))
+            with open(path, 'wb') as fp:
+                fp.write(manifest_msgpack)
+        except Exception:
+            raise
 
     def matching_parse_results(self, manifest: Manifest) -> bool:
         """Compare the global hashes of the read-in parse results' values to
         the known ones, and return if it is ok to re-use the results.
         """
-        try:
-            if manifest.metadata.dbt_version != __version__:
-                logger.debug(
-                    'dbt version mismatch: {} != {}, cache invalidated'
-                    .format(manifest.metadata.dbt_version, __version__)
-                )
-                return False
-        except AttributeError as exc:
-            logger.debug(f"malformed result file, cache invalidated: {exc}")
-            return False
-
         valid = True
 
-        if not self.manifest.state_check or not manifest.state_check:
-            return False
-
+        if manifest.metadata.dbt_version != __version__:
+            logger.info("Unable to do partial parsing because of a dbt version mismatch")
+            return False  # If the version is wrong, the other checks might not work
         if self.manifest.state_check.vars_hash != manifest.state_check.vars_hash:
-            logger.debug('vars hash mismatch, cache invalidated')
+            logger.info("Unable to do partial parsing because config vars, "
+                        "config profile, or config target have changed")
             valid = False
         if self.manifest.state_check.profile_hash != manifest.state_check.profile_hash:
-            logger.debug('profile hash mismatch, cache invalidated')
+            # Note: This should be made more granular. We shouldn't need to invalidate
+            # partial parsing if a non-used profile section has changed.
+            logger.info("Unable to do partial parsing because profile has changed")
             valid = False
 
         missing_keys = {
             k for k in self.manifest.state_check.project_hashes
             if k not in manifest.state_check.project_hashes
         }
         if missing_keys:
-            logger.debug(
-                'project hash mismatch: values missing, cache invalidated: {}'
-                .format(missing_keys)
-            )
+            logger.info("Unable to do partial parsing because a project dependency has been added")
```

Comment:
So glad to have info messages like this one!

Would it be possible (and simple) to do a full comparison between the `project_hashes`, rather than just comparing `k in self.manifest`?

Example: If I had a package installed, parsed my project, manually deleted the installed package files from `dbt_modules`, I'll see a key error for the now-missing file (e.g. `'dbt_utils://macros/cross_db_utils/intersect.sql'`). This is definitely a corner case, since users almost never manually remove installed modules, they're much more likely to run `dbt clean` (which _also_ removes `partial_parse.msgpack`).